src/PereViader.Scrolller.Test/StreamingScrolllerServiceTests.cs

```diff
@@ -0,0 +1,120 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace PereViader.Scrolller.Test;
+
+public class StreamingScrolllerServiceTests
+{
+    private IStreamingScrolllerService _streamingScrolller;
+    private CancellationTokenSource _tokenSource;
+    
+    [SetUp]
+    public void Setup()
+    {
+        var serviceProvider = new ServiceCollection()
+            .AddScrolller()
+            .BuildServiceProvider();
+        _tokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(15));
+
+        _streamingScrolller = serviceProvider.GetRequiredService<IStreamingScrolllerService>();
+    }
+    
+    [Test]
+    [TestCase("humor")]
+    [TestCase("nsfw")]
+    public async Task ScrapeSubreddit_ShouldScrapeSomeUrls(string subreddit)
+    {
+        const int MAX_URLS = 100;
+        var assets = new List<Uri>();
+        try
+        {
+            await foreach (var imageUrl in _streamingScrolller.SubredditUrls(subreddit, _tokenSource.Token))
+            {
+                assets.Add(imageUrl);
+                Assert.That(imageUrl.Scheme, Is.EqualTo(Uri.UriSchemeHttps));
+                
+                if (assets.Count >= MAX_URLS)
+                {
+                    _tokenSource.Cancel();
+                }
+            }
+        }
+        catch (OperationCanceledException)
+        {
+            Assert.That(assets.Count, Is.EqualTo(MAX_URLS));
+        }
```

Comment:
You're right we will not throw an exception however it does get thrown if we ever make another async request. I removed it because I realized we can just leave early